### PR TITLE
fix documentation for write_gml()

### PR DESCRIPTION
**Pull request by SanketDG**
As discussed in #1449 and then stated in #1454.

### CODE REVIEW

File: networkx/readwrite/gml.py [hunk 1]
```diff
@@ -683,10 +683,13 @@ def write_gml(G, path, stringizer=None):
         .bz2 will be compressed.
 
     stringizer : callable, optional
-        A stringizer which converts non-int/float/dict values into strings. If
-        it cannot convert a value into a string, it should raise a
+        A stringizer which converts non-int/non-float/non-dict values into
+        strings. If it cannot convert a value into a string, it should raise a
         ``ValueError`` raised to indicate that. Default value: ``None``.
```
Comment (debugger22) [inline networkx/readwrite/gml.py]: This sounds odd to me "it should raise a `ValueError` raised to indicate that".

Comment (SanketDG) [inline networkx/readwrite/gml.py]: What's wrong with it?

File: networkx/readwrite/gml.py [hunk 2]
```diff
@@ -683,9 +683,12 @@ def write_gml(G, path, stringizer=None):
...
     stringizer : callable, optional
-        A stringizer which converts non-int/float/dict values into strings. If
-        it cannot convert a value into a string, it should raise a
-        ``ValueError`` raised to indicate that. Default value: ``None``.
+        A stringizer which converts non-int/non-float/non-dict values into
+        strings. If it cannot convert a value into a string, it should raise a
+        ``ValueError`` to indicate that. Default value: ``None``.
+
+        Currently, there is no available stringizer to convert
+        non-int/non-float/non-dict values to strings.
```
Comment (chebee7i) [inline networkx/readwrite/gml.py]: I suspect this last statement will end up being more confusing than helpful. I'd prefer to leave it out and discuss this point in greater detail in the user documentation (it's probably not appropriate for a docstring). 

There are stringizers available and they will work just fine depending on your use cases. There are not, however, stringizers/destringizer pairs that will guarantee roundtrips across all versions of Python.

**Comment by chebee7i** [comment]
Thanks for looking into this. What I had in mind was a fairly comprehensive discussion of the issues involved, along with examples. This would be in `doc/source/reference/readwrite.gml.rst`. E.g. the documentation might show what happens if you use `stringizer=json.dumps` or `stringizer=pickle.dumps`, and some of the difficulties one encounters when dealing with different Python versions and also when re-importing back into NetworkX. Using `pickle.dumps` might work, but it somewhat loses the benefit of going to a language-independent format in the first place...but again, all this depends on the use case.

For the docstring, I commented on that last few sentences. I think what could be helpful, instead, is just to mention something along the lines of users needing to give some thought into how the exported data should interact with different languages and even different Python versions. Re-importing from gml is also a concern.  For particular use cases, there are perfectly fine solutions, but it is not a decision that NetworkX can make for the user. Possibly, we could point them to the documentation.

**Comment by SanketDG** [comment]
I'll continue working on this. Apologies for not being able to clearly understand the issue.

So I updated the docstring to only include a minor grammar fix. At the last, I am planning to add this:
Stringizers behave differently across different languages and python versions. Refer to `doc/source/reference/readwrite.gml.rst`.

Is this okay?

**Comment by chebee7i** [comment]
That's probably ok, but it should be changed to an actual http link: http://networkx.github.io/documentation/latest/reference/readwrite.gml.html, and only once we get something in there. Presently, there is no such discussion.

Actually, I'm not sure what the best practice is for linking to user documentation, especially with respect to versions. It would be nice if there was a way to automatically have a version appropriate link, but that might be asking for too much. Maybe others have ideas.

**Comment by SanketDG** [comment]
Will this be merged then for the grammar fix?

**Comment by chebee7i** [comment]
Whatever you feel is best. If you think you might want to work on the extended discussion, we can leave this open and wait. If not, we can merge this now but keep #1454 open. Let us know.

**Comment by SanketDG** [comment]
Okay, let's keep this open for a discussion. Thanks!

**Pull request merged**